### PR TITLE
Registration Form / User management

### PR DESCRIPTION
**Pull request by ghost**
Created a Registration form.

### CODE REVIEW

**Review by janisknets — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -6,14 +6,20 @@ import "bootstrap/dist/css/bootstrap.min.css";
...
   render() {
     return (
       <BrowserRouter>
         <Switch>
           <Route exact path="/" component={App} />
-          <Route exact path="/login" component={Login} />
+
+          <Route exact path="/login" component={Login}/>
+          <Route exact path="/register" component={Registration}/>
+
```
Comment (janisknets) [inline src/index.js]: whitespaces

**Review by janisknets — CHANGES_REQUESTED** [review]

File: src/components/Registration/Registration.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
+import React, { Component } from "react";
+import { Button } from "reactstrap";
+import { Link } from "react-router-dom";
+import "./Registration.css";
+
+
+
```
Comment (rokija) [inline src/components/Registration/Registration.js]: remove empty lines

File: src/components/Registration/Registration.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      email: "",
+      username: "",
+      password: "",
+      name: "",
+      surname: "",
+      dateOfBirth: "",
+      passwordVerification: "",
+      message: "",
+    };
+    this.submitData = this.submitData.bind(this);
```
Comment (rokija) [inline src/components/Registration/Registration.js]: remove this and change submitData to an arrow function

File: src/components/Registration/Registration.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+  inputPassword = event => {
+    this.setState({ password: event.target.value });
+  };
+ 
+  confirmPassword = event => {
+    this.setState({ passwordVerification: event.target.value });
+  };
+
+  submitData(event) {
```
Comment (rokija) [inline src/components/Registration/Registration.js]: change to an arrow function

File: src/components/Registration/Registration.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  }
+
+  onInputChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  
+  
+
+
```
Comment (rokija) [inline src/components/Registration/Registration.js:27]: remove empty lines

File: src/components/Registration/Registration.js [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  inputPassword = event => {
+    this.setState({ password: event.target.value });
```
Comment (rokija) [inline src/components/Registration/Registration.js]: doesnt onInputChange do the same thing?

File: src/components/Registration/Registration.js [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  };
+ 
+  confirmPassword = event => {
+    this.setState({ passwordVerification: event.target.value });
```
Comment (rokija) [inline src/components/Registration/Registration.js]: doesnt onInputChange do the same thing?

File: src/components/Registration/Registration.js [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+                type="text"
+                name="username"
+                value={username}
+              />
+            </div>
+
+            <div className="form-group">
+              <label>Please enter your email</label>
+              <input
+                // onChange={this.onInputChange}
```
Comment (rokija) [inline src/components/Registration/Registration.js]: why is it commented out?

File: src/components/Registration/Registration.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+            </div>
+
+            <div className="form-group">
+              <label>Please enter your password</label>
+              <input
+                // onChange={this.onInputChange}
+                className="form-control"
+                type="password"
+                name="password"
+                onChange={this.inputPassword}
```
Comment (rokija) [inline src/components/Registration/Registration.js]: use onInputChange

File: src/components/Registration/Registration.js [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+              <label>Please repeat the password</label>
+              <input
+                // onChange={this.onInputChange}
+                className="form-control"
+                type="password"
+                // minlength="8" 
+                maxlength="12"
+                required
+                name="passwordVerification"
+                onChange={this.confirmPassword}
```
Comment (rokija) [inline src/components/Registration/Registration.js]: use onInputChange

File: src/components/Registration/Registration.js [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+{/* --------------- Need to compare if passwords match with the if statement. Dunno yet how to do that.---------------- */}
+
+
+            <div className="form-group">
+              <label>Please repeat the password</label>
+              <input
+                // onChange={this.onInputChange}
+                className="form-control"
+                type="password"
+                // minlength="8"
```
Comment (rokija) [inline src/components/Registration/Registration.js]: dont leave commented out code

File: src/components/Registration/Registration.js [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+                onChange={this.onInputChange}
+                className="form-control"
+                type="date"
+                name="dateOfBirth"
+                value={dateOfBirth}
+              />
+            </div>
+            </div>
+            <Button type="submit"
+              // onClick={() =>
```
Comment (rokija) [inline src/components/Registration/Registration.js]: delete commented out code

File: src/components/Registration/Registration.js (hunk 6 above)
Comment (ghost) [reply src/components/Registration/Registration.js]: The inputPassword and Confirm password is created for other blocks. If we delete them, than the whole code must be changed.

File: src/components/Registration/Registration.js (hunk 9 above)
Comment (ghost) [reply src/components/Registration/Registration.js]: onChange={this.onInputChange}

**Review by ghost — COMMENTED** [review]
Comments added.

**Comment by janisknets** [comment]
Will be closed together with user management updates, closing without merge

**Pull request closed without merging**